/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache LicenseVersion 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing;
 * software distributed under the License is distributed on an
 * "AS IS" BASISWITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KINDeither express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 *
 */

syntax = "proto3";

package smiles.model;
import "google/protobuf/struct.proto";

message ExperimentalDP {
  string data_product_id = 1;
  optional string parent_data_product_id = 2;
  optional string name = 3;

  optional string mol_id = 4;
  optional string cas_nr = 5;
  optional string smiles = 6;
  optional string smiles_stereo = 7;
  optional string inchi = 8;
  optional google.protobuf.Struct molfile_blob_source = 9;
  optional string emp_formula = 10;
  optional string emp_formula_sort = 11;
  optional string emp_formula_source = 12;
  optional double mw = 13;
  optional double mw_monoiso = 14;
  optional double rdb = 15;
  optional string mw_source = 16;
  optional string validated_by = 17;
  optional string journal = 18;
  optional string auth_of_intr = 19;
  optional string jour_cit = 20;
  optional string year_publ = 21;
  optional string doi_link = 22;
  optional string comp_class = 23;
  optional string cuniq = 24;
  optional string calc_perf = 25;
  optional string org_met = 26;
  optional int64 mol_chrg = 27;
  optional string state_ofmat = 28;
  optional string color_white = 29;
  optional string color_uv = 30;
  optional double absorb_max = 31;
  optional string solvent_ae = 32;
  optional double absorb = 33;
  optional double conc = 34;
  optional double extinc = 35;
  optional double emis_max = 36;
  optional double temp_abs = 37;
  optional double emis_qy = 38;
  optional double temp_ems = 39;
  optional double lifetime = 40;
  optional double temp_cv = 41;
  optional double reduc_pot = 42;
  optional string hw_or_pk_rp = 43;
  optional double oxid_pot = 44;
  optional string hw_or_pk_op = 45;
  optional string solvent_cv = 46;
  optional string electrolyte = 47;
  optional string ref_electrd = 48;
  optional string inter_thngs = 49;
  optional double density_20 = 50;
  optional string density_20_source = 51;
  optional double default_warn_level = 52;
  optional double n_20 = 53;
  optional string n_20_source = 54;
  optional double mp_low = 55;
  optional double mp_high = 56;
  optional string mp_source = 57;
  optional double bp_low = 58;
  optional double bp_high = 59;
  optional double bp_press = 60;
  optional string press_unit = 61;
  optional string bp_source = 62;
  optional string safety_r = 63;
  optional string safety_h = 64;
  optional string safety_s = 65;
  optional string safety_p = 66;
  optional string safety_text = 67;
  optional string safety_sym = 68;
  optional string safety_sym_ghs = 69;
  optional string safety_source = 70;
  optional string comment_mol = 71;
}

